script.embuary.info/resources/lib/tmdb_main.py

```diff
@@ -0,0 +1,333 @@
+        self.cont3.addItems(self.images)
+
+    def onAction(self, action):
+        if action.getId() in [92,10]:
+            self.action['id'] = ''
+            self.action['call'] = 'back' if action.getId() == 92 else 'close'
+            self.quit()
+
+    def onClick(self,controlId):
+        next_id = xbmc.getInfoLabel('Container(%s).ListItem.Property(id)' % controlId)
+        next_call = xbmc.getInfoLabel('Container(%s).ListItem.Property(call)' % controlId)
+
+        if next_call in ['person','movie','tv'] and next_id:
+            self.action['id'] = next_id
+            self.action['call'] = next_call
+            self.quit()
+
+        elif next_call == 'image':
+            FullScreenImage(controlId)
+
+    def quit(self):
+        self.close()
+
+
+''' Show & movie dialog
+'''
+class DialogVideo(xbmcgui.WindowXMLDialog):
+    def __init__(self,*args,**kwargs):
+        self.first_load = True
+        self.action = {}
+
+        self.details = kwargs['details']
+        self.cast = kwargs['cast']
+        self.crew = kwargs['crew']
+        self.similar = kwargs['similar']
+        self.youtube = kwargs['youtube']
+        self.backdrops = kwargs['backdrops']
+
+    def __getitem__(self,key):
+        return self.action[key]
+
+    def __setitem__(self,key,value):
+        self.action[key] = value
+
+    def onInit(self):
+        if self.first_load:
+            self.add_items()
+
+    def add_items(self):
+        self.first_load = False
+        self.cont0 = self.getControl(10051)
+        self.cont0.addItems(self.details)
+        self.cont1 = self.getControl(10052)
+        self.cont1.addItems(self.cast)
+        self.cont2 = self.getControl(10053)
+        self.cont2.addItems(self.similar)
+        self.cont3 = self.getControl(10054)
+        self.cont3.addItems(self.youtube)
+        self.cont4 = self.getControl(10055)
+        self.cont4.addItems(self.backdrops)
+        self.cont5 = self.getControl(10056)
+        self.cont5.addItems(self.crew)
+
+    def onAction(self, action):
+        if action.getId() in [92,10]:
+            self.action['id'] = ''
+            self.action['call'] = 'back' if action.getId() == 92 else 'close'
+            self.quit()
+
+    def onClick(self,controlId):
+        next_id = xbmc.getInfoLabel('Container(%s).ListItem.Property(id)' % controlId)
+        next_call = xbmc.getInfoLabel('Container(%s).ListItem.Property(call)' % controlId)
+
+        if next_call in ['person','movie','tv'] and next_id:
+            self.action['id'] = next_id
+            self.action['call'] = next_call
+            self.quit()
+
+        elif next_call == 'image':
+            FullScreenImage(controlId)
+
+        elif next_call == 'youtube':
+            self.action['call'] = 'close'
+            execute('Dialog.Close(all)')
+            execute('PlayMedia(plugin://plugin.video.youtube/play/?video_id=%s)' % xbmc.getInfoLabel('Container(%s).ListItem.Property(ytid)' % controlId))
```

Comment:
This will likely crash Kodi. Use xbmc.Player().play()

Reply:
Never had a crash using this, but I'll change it

Reply:
It's a know issue in Kodi for calls that activate the busydialog.

Reply:
changed